src/plugins/teleop/Teleop.cc

```diff
@@ -0,0 +1,272 @@
+
+    /// \brief Linear direction.
+    public: int linearDir = 0;
+    /// \brief Angular direction.
+    public: int angularDir = 0;
+
+    /// \brief Linear state setted by keyboard input.
+    public: keyLinear linearState = keyLinear::kstop;
+    /// \brief Angular state setted by keyboard input.
+    public: keyAngular angularState = keyAngular::kstop;
+
+    public: bool keyEnable = false;
+    public: bool newTopic = true;
+
+  };
+}
+}
+}
+
+using namespace ignition;
+using namespace gui;
+using namespace plugins;
+
+/////////////////////////////////////////////////
+Teleop::Teleop(): Plugin(), dataPtr(new TeleopPrivate)
+{
+}
+
+/////////////////////////////////////////////////
+Teleop::~Teleop()
+{
+}
+
+/////////////////////////////////////////////////
+void Teleop::LoadConfig(const tinyxml2::XMLElement *_pluginElement)
+{
+  if (this->title.empty())
+    this->title = "Teleop";
+
+  ignition::gui::App()->findChild
+    <ignition::gui::MainWindow *>()->QuickWindow()->installEventFilter(this);
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnTeleopTwist()
+{
+  ignition::msgs::Twist cmdVelMsg;
+
+  cmdVelMsg.mutable_linear()->set_x(
+      this->dataPtr->linearDir * this->dataPtr->linearVel);
+  cmdVelMsg.mutable_angular()->set_z(
+      this->dataPtr->angularDir * this->dataPtr->angularVel);
+
+  if (this->dataPtr->newTopic){
+    this->dataPtr->cmdVelPub = ignition::transport::Node::Publisher();
+    this->dataPtr->cmdVelPub =
+    this->dataPtr->node.Advertise<ignition::msgs::Twist>(this->dataPtr->topic);
+    this->dataPtr->newTopic = false;
+    this->dataPtr->cmdVelPub.Publish(cmdVelMsg);
+  }
+
+  this->dataPtr->cmdVelPub.Publish(cmdVelMsg);
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnTopicSelection(const QString& _topic)
+{
+  this->dataPtr->newTopic = true;
+  this->dataPtr->topic = _topic.toStdString();
+  ignmsg << "A new topic has been entered: '" <<
+      this->dataPtr->topic << " ' " <<std::endl;
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnLinearVelSelection(const QString& _velocity)
+{
+  this->dataPtr->linearVel = _velocity.toDouble();
+  ignmsg << "[OnlinearVelSelection]: linear velocity: "
+      << this->dataPtr->linearVel << std::endl;
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnAngularVelSelection(const QString& _velocity)
+{
+  this->dataPtr->angularVel = _velocity.toDouble();
+  ignmsg << "[OnlinearVelSelection]: angular velocity: "
+      << this->dataPtr->angularVel << std::endl;
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnKeySwitch(bool _checked)
+{
+  this->dataPtr->linearDir = 0;
+  this->dataPtr->angularDir = 0;
+  this->dataPtr->keyEnable = _checked;
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnSlidersSwitch(bool _checked)
+{
+  if(_checked){
+    this->dataPtr->linearDir = 1;
+    this->dataPtr->angularDir = 1;
+    OnTeleopTwist();
```

Comment:
Style: always use `this`. There are other instances

```suggestion
    this->OnTeleopTwist();
```

Reply:
Addressed